internal/security/secretstore/tokenprovider.go

```diff
@@ -0,0 +1,94 @@
+//
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under
+// the License.
+//
+// SPDX-License-Identifier: Apache-2.0'
+//
+
+package secretstore
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"strings"
+	"syscall"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+)
+
+const OneShotProvider = "oneshot"
+
+type TokenProvider struct {
+	logging         logger.LoggingClient
+	ctx             context.Context
+	initialized     bool
+	config          SecretServiceInfo
+	resolvedPath    string
+	oneShotProvider bool
+}
+
+// NewTokenProvider creates a new TokenProvider
+func NewTokenProvider(ctx context.Context) *TokenProvider {
+	return &TokenProvider{
+		logging: LoggingClient,
+		ctx:     ctx,
+	}
+}
+
+func (p *TokenProvider) SetConfiguration(config SecretServiceInfo) error {
+	var err error
+	p.config = config
+	if p.config.TokenProviderType != OneShotProvider {
+		err := fmt.Errorf("%s is not a supported TokenProviderType", p.config.TokenProviderType)
+		p.logging.Error(err.Error())
+		return err
+	}
+	p.resolvedPath, err = exec.LookPath(p.config.TokenProvider)
+	if err != nil {
+		p.logging.Error(fmt.Sprintf("Failed to locate %s on PATH: %s", p.config.TokenProvider, err.Error()))
+		return err
+	}
+	p.oneShotProvider = (p.config.TokenProviderType == OneShotProvider)
```

Comment:
Do I miss something- why compares type again? It looks like it will be always true when the code reaches here.